lib/js_rails_routes/generator.rb

```diff
@@ -18,24 +18,29 @@ class Generator
 
     include Singleton
 
-    attr_accessor :include_paths, :exclude_paths, :include_names, :exclude_names, :path
+    attr_accessor :include_paths, :exclude_paths, :include_names, :exclude_names, :exclude_engines, :base_path
 
     def initialize
-      self.include_paths = /.*/
-      self.exclude_paths = /^$/
-      self.include_names = /.*/
-      self.exclude_names = /^$/
-      self.path = Rails.root.join('app', 'assets', 'javascripts', 'rails-routes.js')
+      self.include_paths   = /.*/
+      self.exclude_paths   = /^$/
+      self.include_names   = /.*/
+      self.exclude_names   = /^$/
+      self.exclude_engines = /^$/
+
+      self.base_path = Rails.root.join('app', 'assets', 'javascripts')
       Rails.application.reload_routes!
     end
 
     def generate(task)
-      lines = ["// Don't edit manually. `rake #{task}` generates this file.", PROCESS_FUNC]
-      lines += routes.map do |route_name, route_path|
-        handle_route(route_name, route_path) if match?(route_name, route_path)
-      end.compact
-      lines += [''] # End with new line
-      write(lines.join("\n"))
+      routes_with_engines.each do |klass, routes|
```

Comment:
Please  rename`klass` as `rails_engine_name`

Reply:
Make sense. it's a string.